pilot/pkg/model/push_context.go

```diff
@@ -495,3 +499,10 @@ func (ps *PushContext) SubsetToLabels(subsetName string, hostname Hostname) Labe
 
 	return nil
 }
+
+func deepCopyVirtualService(in *networking.VirtualService) *networking.VirtualService {
+	out := new(networking.VirtualService)
+	data, _ := json.Marshal(in)
+	json.Unmarshal(data, out)
```

Comment:
Here returns an error `json: cannot unmarshal object into Go struct field StringMatch.MatchType of type v1alpha3.isStringMatch_MatchType`

So I can not use this way to deepcopy

Reply:
Deep copy via json is a performance killler FYI